packages/sit-onyx/src/components/OnyxRadioButtonGroup/OnyxRadioButtonGroup.vue

```diff
@@ -1,39 +1,26 @@
-<script lang="ts" setup generic="T">
+<script lang="ts" setup generic="TValue">
 import type { TargetEvent } from "@/types/dom";
 import OnyxRadioButton from "../OnyxRadioButton/OnyxRadioButton.vue";
 import type { SelectionOption } from "../OnyxRadioButton/types";
 import OnyxHeadline from "../OnyxHeadline/OnyxHeadline.vue";
+import type { OnyxRadioButtonGroupProps } from "./types";
+import { createId } from "@sit-onyx/headless";
 
 type ChangeEvent = TargetEvent<HTMLInputElement>;
 
-const props = defineProps<{
-  /**
-   * Name for the radio button group form element.
-   * Warning: Never use a name for form elements twice!
-   */
-  name: string;
-  modelValue?: SelectionOption<T>;
-  /**
-   * Headline shown above the radio button group, which is also the fieldset legend.
-   */
-  headline?: string;
-  required?: boolean;
-  disabled?: boolean;
-  readonly?: boolean;
-  isLoading?: boolean;
-  errorMessage?: string;
-  options: SelectionOption<T>[];
-}>();
+const props = withDefaults(defineProps<OnyxRadioButtonGroupProps<TValue>>(), {
+  name: createId("radio-button-group-name"), // the name must be globally unique
```

Comment:
Why not make the name required?

Reply:
The name is only needed for technical reasons:
- linking the radio buttons together
- name for the key of the respective form data entry

So most of the time it doesn't matter what the name is aslong as it is unique.
The `name` prop being optional makes it easier to use for the developers.